app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt

```diff
@@ -71,7 +68,8 @@ fun CreateAChallengeScreen(
                               selectedFriendId = friendId
                               showDialog = true
                             },
-                            modifier = Modifier.padding(8.dp)) {
+                            modifier =
+                                Modifier.padding(8.dp).testTag("ChallengeButton_$friendId")) {
                               Text("Challenge")
                             }
                       }
```

Comment:
The dialog design is clean and functional. It might be helpful to add visual feedback (e.g., a highlight or animation) when a mode is selected to improve clarity.